R/utils-get_code_dependency.R

```diff
@@ -35,17 +35,22 @@ get_code_dependency <- function(code, names, check_names = TRUE) {
 
   code <- parse(text = code, keep.source = TRUE)
   pd <- utils::getParseData(code)
+  calls_pd <- extract_calls(pd)
 
   if (check_names) {
     # Detect if names are actually in code.
-    symbols <- unique(pd[pd$token == "SYMBOL", "text"])
-    if (!all(names %in% symbols)) {
+    symbols <- unlist(lapply(calls_pd, function(call) call[call$token == "SYMBOL", "text"]))
+    if (any(pd$text == "assign")) {
+      assign_calls <- Filter(function(call) any(call$token == "SYMBOL_FUNCTION_CALL" & call$text == "assign"), calls_pd)
```

Comment:
Consider creating a function like `is_assign` to use here and then again in `extract_occurence`.

Reply:
Great improvement. I created `is_symbol` function and also used that in situation where we seek for `data` function as well b56c5c426c96dac78130a395a7014e16b2b4d369